.github/workflows/autobuild.yml

```diff
@@ -111,22 +111,24 @@ jobs:
 
            - config_name:           MacOS (codeQL)
              target_os:             macos
-             building_on_os:        macos-10.15
+             building_on_os:        macos-11
              cmd1_prebuild:         "./autobuild/mac/codeQL/autobuild_mac_1_prepare.sh 5.15.2"
              cmd2_build:            "./autobuild/mac/codeQL/autobuild_mac_2_build.sh"
              cmd3_postbuild:        false
              uses_codeql:           true
-             xcode_version:         12.1.1
+             xcode_version:         12.5.1
 
            - config_name:           MacOS (artifacts)
              target_os:             macos
-             # Stay on 10.15 as long as we use dmgbuild which does not work with 11's hdiutil (?)
-             building_on_os:        macos-10.15
+             building_on_os:        macos-11
              cmd1_prebuild:         "./autobuild/mac/artifacts/autobuild_mac_1_prepare.sh 5.15.2"
              cmd2_build:            "./autobuild/mac/artifacts/autobuild_mac_2_build.sh sign_if_possible"
              cmd3_postbuild:        "./autobuild/mac/artifacts/autobuild_mac_3_copy_files.sh"
              uses_codeql:           false
-             xcode_version:         12.1.1
+             # Qt5's maximum supported Xcode version is 12.x (SDK 11). Therefore, use the latest
+             # 12.x release here.
+             # https://xcodereleases.com/ https://doc.qt.io/qt-5/macos.html
```

Comment:
I see there is still a build warning:

```
Project WARNING: Qt has only been tested with version 10.15 of the platform SDK, you're using 11.3.
Project WARNING: This is an unsupported configuration. You may experience build issues, and by using
Project WARNING: the 11.3 SDK you are opting in to new features that Qt has not been prepared for.
Project WARNING: Please downgrade the SDK you use to build your app to version 10.15, or configure
Project WARNING: with CONFIG+=sdk_no_version_check when running qmake to silence this warning.
```

That is at odds with what is stated at https://doc.qt.io/qt-5/macos.html

It makes me wonder whether SDK11 compatibility was introduced in one of the proprietary Qt 5.15.x where x > 2?

If thorough testing shows the built artefact does indeed run properly, maybe we should add the specified `CONFIG+=sdk_no_version_check` when building this version?

Reply:
> I see there is still a build warning:

Thanks, I've missed this as I was relying on the documented behavior.

> It makes me wonder whether SDK11 compatibility was introduced in one of the proprietary Qt 5.15.x where x > 2?

Possible, yes.

> If thorough testing shows the built artefact does indeed run properly, maybe we should add the specified CONFIG+=sdk_no_version_check when building this version?

Yes, if we continue this path we should definitely add this and explain why we did it. However, as long as there are no positive reports about the Xcode update fixing things, this warning reinforced my tendency not to perform this change right now.

Therefore, I'm going to close this PR. If we have reports that it helps, I'll reopen and add the `sdk_no_version_check`.
I'll open a seperate PR for `macos-11` as we should do that anyway.